### PR TITLE
closes #24442: refactor recent tabs to use common composables

### LINKED ISSUE #24442: Refactor recent tabs to use generic composables.
As prep for landing #24396, the following composables in `view/RecentTabs` should be replaced with generic composables:

- `RecentTabImage`
- `RecentTabTitle`
- `RecentTabSubtitle`

### PR DESCRIPTION
**Pull request by MatthewTighe**
closes #24442

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/compose/ThumbnailCard.kt [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import mozilla.components.browser.icons.compose.Loader
+import mozilla.components.browser.icons.compose.Placeholder
+import mozilla.components.browser.icons.compose.WithIcon
+import mozilla.components.concept.base.images.ImageLoadRequest
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.components
+import org.mozilla.fenix.theme.FirefoxTheme
+
+@Composable
+fun ThumbnailCard(
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/compose/ThumbnailCard.kt:36]: kDoc please

File: app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt [hunk 2]
```diff
@@ -413,43 +430,7 @@ private fun RecentTabImage(
...
-
-                ThumbnailImage(
-                    key = key,
-                    modifier = modifier,
-                    contentScale = contentScale,
-                    alignment = alignment
-                )
-            }
-        }
+        else -> ThumbnailCard(tab.state.content.url, tab.state.id, modifier)
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt:433]: Can you add the named parameters and make it multiline

**Review by gabrielluong — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/compose/ThumbnailCard.kt [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+ * will be displayed until the thumbnail is loaded.
+ *
+ * @param url Url to display thumbnail for.
+ * @param key Key used to remember the thumbnail for future compositions.
+ * @param modifier [Modifier] used to draw the image content.
+ * @param contentScale [ContentScale] used to draw image content.
+ * @param alignment [Alignment] used to draw the image content.
+ */
+@Composable
+fun ThumbnailCard(
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/compose/ThumbnailCard.kt:46]: can we get a preview for this? (this can be follow-up)

**Comment by MatthewTighe** [comment]
Thanks, trying to sort that out. Test passes locally for me 🤷

**Pull request merged**